patroni/postgresql.py

```diff
@@ -1058,11 +1084,18 @@ def _write_postgresql_conf(self):
         with open(self._postgresql_conf, 'w') as f:
             f.write('# Do not edit this file manually!\n# It will be overwritten by Patroni!\n')
             f.write("include '{0}'\n\n".format(self.config.get('custom_conf') or self._postgresql_base_conf_name))
-            f.write("data_directory = '{}'\n".format(self._data_dir))
-            f.write("hba_file = '{}'\n".format(os.path.join(self._config_dir, 'pg_hba.conf')))
-            f.write("ident_file = '{}'\n".format(os.path.join(self._config_dir, 'pg_ident.conf')))
+            f.write("data_directory = '{0}'\n".format(self._data_dir))
             for name, value in sorted(self._server_parameters.items()):
-                f.write("{0} = '{1}'\n".format(name, value))
+                if not self._running_custom_bootstrap or name != 'hba_file':
+                    f.write("{0} = '{1}'\n".format(name, value))
+            # when we are doing custom bootstrap we assume that we don't know superuser passwrod
+            # and in order to be able to change it, we are opening trust access from a sertain address
+            # therefore we need to make sure that hba_file is not overriden
+            # after changing superuser password we will "revert" all these "changes"
+            if self._running_custom_bootstrap or 'hba_file' not in self._server_parameters:
+                f.write("hba_file = '{0}'\n".format(self._pg_hba_conf))
+            if 'ident_file' not in self._server_parameters:
+                f.write("ident_file = '{0}'\n".format(os.path.join(self._config_dir, 'pg_ident.conf')))
```

Comment:
It still needs to be there. Actually we never tried to backup it and postgres doesn't like it, i.e. shows warning.